File: Data Structures & Algorithms/merge-intervals/submission-0.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end()); // sorted by start time
        vector<vector<int>> res;

        for( auto& interval : intervals){
            if(res.empty() || res.back()[1] < interval[0]){
                res.push_back(interval);
            } else{
                res.back()[1] = max(res.back()[1], interval[1]);
            }
        }
        return res; 

    }
};
        /*
        sort = [[1,3],[1,5],[6,7]]
        curr_int = [1,3]
        next= [1,5] 1 <= 3
        merge = [1,5]
        next = [6,7] 6 <= 5


        
        */

File: Data Structures & Algorithms/non-overlapping-intervals/submission-0.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort()
        res = 0
        prvEnd = intervals[0][1] #first interval end time

        for start, end in intervals[1:]:
            if start >= prvEnd:
                prvEnd = end
            else:
                res += 1
                prvEnd = min(end, prvEnd)
        return res